File: ios-native/CallBrandingPlugin.swift
//
//  CallBrandingPlugin.swift
//  SecureNode Test
//
//  Capacitor plugin for Call Branding
//

import Foundation
import Capacitor

/**
 * Capacitor plugin to bridge JavaScript to native call branding
 */
@objc(CallBrandingPlugin)
public class CallBrandingPlugin: CAPPlugin {
    
    @objc func initialize(_ call: CAPPluginCall) {
        guard let apiKey = call.getString("apiKey") else {
            call.reject("API key is required")
            return
        }
        
        let apiUrl = call.getString("apiUrl") ?? "https://calls.securenode.io/api"
        
        CallKitManager.shared.initialize(apiKey: apiKey, apiUrl: apiUrl)
        
        // Store API key for persistence
        UserDefaults.standard.set(apiKey, forKey: "securenode_api_key")
        UserDefaults.standard.set(apiUrl, forKey: "securenode_api_url")
        
        call.resolve()
    }
    
    @objc func syncBranding(_ call: CAPPluginCall) {
        CallKitManager.shared.syncBrandingData()
        call.resolve([
            "success": true
        ])
    }
    
    @objc func testIncomingCall(_ call: CAPPluginCall) {
        guard let phoneNumber = call.getString("phoneNumber") else {
            call.reject("Phone number is required")
            return
        }
        
        let uuid = UUID()
        CallKitManager.shared.handleIncomingCall(uuid: uuid, phoneNumber: phoneNumber)
        
        call.resolve()
    }
}

File: android-native/CallBrandingPlugin.java
package io.securenode.nodekit;

import android.util.Log;
import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

/**
 * Capacitor plugin for Call Branding
 */
@CapacitorPlugin(name = "CallBranding")
public class CallBrandingPlugin extends Plugin {
    
    private static final String TAG = "CallBrandingPlugin";
    
    @PluginMethod
    public void initialize(PluginCall call) {
        String apiKey = call.getString("apiKey");
        if (apiKey == null || apiKey.isEmpty()) {
            call.reject("API key is required");
            return;
        }
        
        String apiUrl = call.getString("apiUrl", "https://calls.securenode.io/api");
        
        // Initialize ConnectionService
        SecureNodeConnectionService.initialize(
            getContext(),
            apiKey,
            apiUrl
        );
        
        // Store API key for persistence
        getContext().getSharedPreferences("SecureNodePrefs", 0)
            .edit()
            .putString("api_key", apiKey)
            .putString("api_url", apiUrl)
            .apply();
        
        call.resolve();
    }
    
    @PluginMethod
    public void syncBranding(PluginCall call) {
        // Sync is handled by the SDK automatically
        // This method can trigger a manual sync if needed
        JSObject result = new JSObject();
        result.put("success", true);
        call.resolve(result);
    }
    
    @PluginMethod
    public void testIncomingCall(PluginCall call) {
        String phoneNumber = call.getString("phoneNumber");
        if (phoneNumber == null || phoneNumber.isEmpty()) {
            call.reject("Phone number is required");
            return;
        }
        
        // For testing, we can simulate an incoming call
        // In production, this would come from the system
        Log.d(TAG, "Test incoming call: " + phoneNumber);
        
        // Note: Actual call handling is done by ConnectionService
        // This is just for testing/development
        call.resolve();
    }
}

